server/asset/bch/live_test.go

```diff
@@ -63,15 +58,16 @@ func TestMain(m *testing.M) {
 		var ok bool
 		bch, ok = dexAsset.(*BCHBackend)
 		if !ok {
-			fmt.Printf("Could not cast asset.Backend to *Backend")
+			fmt.Printf("Could not cast asset.Backend to *BCHBackend")
 			return 1
 		}
 
-		wg, err = dexAsset.Connect(ctx)
+		wg, err := dexAsset.Connect(ctx)
 		if err != nil {
 			fmt.Printf("Connect failed: %v", err)
 			return 1
 		}
+		defer wg.Wait()
```

Comment:
Look like it's going to start waiting before cancelling now when `m.Run()` completes because this defer is on the top of the defer stack.  When looking at the previous revision where I had requested a change, I had envisioned moving the `ctx, cancel = context.WithCancel(context.Background())` down to right before `dexAsset.Connect(ctx)` because `ctx` isn't used before that, thus allowing the `cancel(); wg.Wait()` to go in the same closure again, just down here.